tests/emurt/emurt.c

```diff
@@ -1,54 +1,19 @@
 #include "emurt.h"
 
+#define EXIT_REG 0x10000000
+#define UART_W_REG 0x10000010
+#define PERF_REG 0x10000014
+
+void t1_put_char(char c) { *(uint32_t *)(UART_W_REG) = (uint32_t)c; }
+void place_counter(int i) { *(int *)(PERF_REG) = i; }
```

Comment:
Please use `volatile` for mmio access.

Even RV psabi defined `char` is unsigned, `char` is signed on some arch (e.g. x86 gcc). I suggest change to `(uint8_t)c` to make the conversion more non-ambiguous.

Reply:
Should be fixed, please review again.

Reply:
LGTM now